### ISSUE TITLE
Accented characters in math failed to render correctly in Safari

### ISSUE THREAD

**Issue by murattasdemir** [quarto-dev/quarto-cli#5636]
### Bug description

Quarto fails to render accents such as tilde and hat in latex equations correctly in revealjs presentation. Below I provide a mwe.
<img width="58" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/40790433/2bcd0713-7d21-4127-a913-5617a0e34ae4">

Safari 16.0, macOS 12.6, MacBook Pro M1 chip. Quarto 1.3.340. Rendered using VS code Quarto extension.

```{qmd}
title: "Minimal example"
format: 
  revealjs:
    theme: [default]
---
## Minimal example

- Tilde problem with Safari: $\tilde c$ or $\hat{c}$
```
`quarto check` output:

```
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.3.340
      Path: /Applications/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.9.13 (Conda)
      Path: /opt/homebrew/anaconda3/bin/python
      Jupyter: 4.11.1
      Kernels: ir, julia-1.8, python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.2.0
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/library
      knitr: 1.42
      rmarkdown: 2.19

[✓] Checking Knitr engine render......OK
```

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.
- [X] Please provide the output of `quarto check` so we know which version of quarto and its dependencies you're running.

**Comment by cderv** [comment]
@dragonstyle do you see the issue on Safari ? I don't have Safari available on Windows I believe. So I can't really look into that. 

Everything looks ok to me on windows with Chrome. 
![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/11d442c5-a21a-4ca8-8f5c-64732fbf8ab6)

It is probably a font issue for those characters ... but Mathjax should bring the font I believe. 

I found old issues with Mathjax, tilde and safari keyword but quite old https://github.com/mathjax/MathJax/issues/1737
It seems they ruled it out as a browser bug. 🤔

**Comment by mcanouil** [comment]
The issue does not seem to be Safari, since in VSCode chromium browser, I see the same behaviour on macOS ventura 13.4.
Also, this behaviour is when using `revealjs`.

| `html` | `revealjs` |
|:-:|:-:|
| <img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/89164911-f10a-43cb-8cd6-62c64007fb5b"> | <img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/c337edcd-364a-44e5-bd6a-21b0f4e339f1"> |

Something in the CSS (Quarto or Reveal.js) messes up the math display as it is fixed when changing rendering mode.
<img width="795" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/89772aef-de20-4526-b09e-0c9bf4f66367">

**Comment by mcanouil** [comment]
Additionally, the "first" rendering is correct, then something happens.

https://github.com/quarto-dev/quarto-cli/assets/8896044/5c80fce7-23a5-493e-ab5a-5b3518899124

**Comment by cderv** [comment]
Oh really interesting ! Thanks for looking into this. 

It could be related to Mathjax 2.7 as revealjs still defaults to 3 (https://github.com/quarto-dev/quarto-cli/issues/4493) 

The issue I found on Mathjax is with 2.7 BTW. 

> then something happens.

This when Mathjax happens IMO. So possibly a bug with Mathjax 2.7 

If so, it would probably be the same issue with **rmarkdown** or using Mathjax in a HTML doc itself. We could try checking this. 

> Something in the CSS (Quarto or Reveal.js) messes up the math display as it is fixed when changing rendering mode.

I think it is something inside Mathjax itself. HTML-CSS is a rendering method within Mathjax. From https://docs.mathjax.org/en/v2.7-latest/output.html

> The HTML-CSS output processor produces high-quality output in all browsers, with results that are consistent across browsers and operating systems. This was MathJax’s primary output mode prior to MathJax v2.6. Its major advantage is its quality and consistency on legacy browsers starting with IE6. **The HTML-CSS output uses web-based fonts so that users don’t have to have math fonts installed on their computers but can use locally installed fonts instead**.

So I would think this is a font issues, and probably on Mac Only then as I can't reproduce

**Comment by mcanouil** [comment]
> I think it is something inside Mathjax itself. HTML-CSS is a rendering method within Mathjax.

I did a quick look-up at the HTML/CSS produced and to me it seems to be a third party issue rather than a Quarto issue.

**Comment by cderv** [comment]
> What's weird, is that the font is changed during laoding.

I think it is expected as Mathjax will use its fonts.

**Comment by mcanouil** [comment]
I confirm MathJax 3 is the solution.
<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/892edf04-50f3-4b39-bac5-fc9c4aa56704">

**Comment by cderv** [comment]
> I confirm MathJax 3 is the solution.

Good. We just need to adapt for the revealMath plugin - Pandoc is still broken on this and we need to patch the template probably.

I will close this issue and #4493 is to track
